packages/native/src/components/Carousel/index.tsx

```diff
@@ -42,15 +54,28 @@ export type Props = React.PropsWithChildren<{
    * This container is a Flex element.
    */
   slideIndicatorContainerProps?: FlexboxProps & ViewProps;
+
+  IndicatorComponent?:
+    | React.ComponentType<{
+        activeIndex: number;
+        slidesLength: number;
+        onChange?: (index: number) => void;
+        duration?: number;
+      }>
+    | React.ReactElement;
```

Comment:
not a big fan of mixed types, it works but it requires a probably costly check on `React.isValidElement` where you could just have an extra prop `IndicatorElement`
and check what to show 
```js
{
 IndicatorElement ? (
          IndicatorElement
        ) : IndicatorComponent 
         ? (
          <IndicatorComponent
            ...
          />
        ) : null
 }
```